File: api/MfaApi/src/Modules/Member/Contracts/GetMembersByDateResponse.cs
using MfaApi.Modules.Membership;

namespace MfaApi.Modules.Member;

public class GetMembersByDateResponse {
    public Guid Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public DateOnly JoinedDate { get; set; }
    public Guid MembershipId { get; set; }
    public MembershipDto? Membership { get; set; }

    public class MembershipDto {
        public required Guid Id { get; set; }
        public required MembershipType MembershipType { get; set; }
    }
}

File: dashboard/src/modules/stats/types/get-members-by-date-response.ts
import { MembershipType } from '@/modules/membership/types';

export interface GetMembersByDateResponse {
  id: string;
  firstName: string;
  lastName: string;
  joinedDate: Date;
  membershipId: string;
  membership: {
    membershipId: string;
    membershipType: MembershipType;
  }
}